src/main/java/frc/robot/subsystems/conveyer/Conveyer.java

```diff
@@ -0,0 +1,144 @@
+
+public class Conveyer extends SubsystemBase {
+  private CANSparkMax frontMotor =
+      new CANSparkMax(RobotMap.FRONT_CONVEYER_CAN_ID, MotorType.kBrushless);
+  private CANSparkMax backMotor =
+      new CANSparkMax(RobotMap.BACK_CONVEYER_CAN_ID, MotorType.kBrushless);
+
+  private RelativeEncoder frontMotorEncoder = frontMotor.getEncoder();
+  private RelativeEncoder backMotorEncoder = backMotor.getEncoder();
+
+  /**
+   * These aren't actually limit switches; we just use SparkLimitSwitch objects to access them
+   * easily. At the moment, these are unused.
+   */
+  SparkLimitSwitch
+      beamBreakSensorOne = frontMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen),
+      beamBreakSensorTwo = frontMotor.getReverseLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen),
+      beamBreakSensorThree = backMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen),
+      beamBreakSensorFour = backMotor.getReverseLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+
+  /**
+   * Used to define the current position of the note.
+   *
+   * <ul>
+   *   <li>NO_NOTE: There is no note in the conveyer.
+   *   <li>HOLDING_NOTE: The conveyer is holding a note in place.
+   *   <li>PARTIAL_NOTE: The conveyer is in the process of receiving a note from the intake, sending
+   *       a note to the shooter, or scoring a note in the trap or amp.
+   * </ul>
+   */
+  public enum ConveyerPositions {
+    NO_NOTE,
+    HOLDING_NOTE,
+    PARTIAL_NOTE
+  }
+
+  public ConveyerPositions currentNotePosition = ConveyerPositions.NO_NOTE;
+
+  private double backMotorZeroPosition;
+
+  public Conveyer() {
+    SparkMaxUtils.initWithRetry(this::setUpConveyerSparks, ConveyerCal.SPARK_INIT_RETRY_ATTEMPTS);
+    backMotorZeroPosition = backMotorEncoder.getPosition();
+  }
+
+  /** Does all the initialization for the sparks, return true on success */
+  private boolean setUpConveyerSparks() {
+    int errors = 0;
+    errors += SparkMaxUtils.check(frontMotor.restoreFactoryDefaults());
+    errors += SparkMaxUtils.check(backMotor.restoreFactoryDefaults());
+
+    errors += SparkMaxUtils.check(frontMotor.setIdleMode(IdleMode.kBrake));
+    errors += SparkMaxUtils.check(backMotor.setIdleMode(IdleMode.kBrake));
+
+    frontMotor.setInverted(ConveyerConstants.FRONT_MOTOR_INVERTED);
+    backMotor.setInverted(ConveyerConstants.BACK_MOTOR_INVERTED);
+
+    errors +=
+        SparkMaxUtils.check(
+            frontMotor.setSmartCurrentLimit(ConveyerCal.CONVEYER_CURRENT_LIMIT_AMPS));
+    errors +=
+        SparkMaxUtils.check(
+            backMotor.setSmartCurrentLimit(ConveyerCal.CONVEYER_CURRENT_LIMIT_AMPS));
+
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setDegreesFromGearRatio(frontMotorEncoder, ConveyerConstants.FRONT_GEAR_RATIO));
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setDegreesFromGearRatio(backMotorEncoder, ConveyerConstants.BACK_GEAR_RATIO));
```

Comment:
Should we set these to linear? I landed some stuff for setting these to linear for the elevator